### DISCUSSION TITLE
Wich branch may be used as a source for contributing ?

### DISCUSSION THREAD

**Discussion by DzzD** [supermerill/SuperSlicer#2831] [Q&A] [answered]
So here is my question, is there a prefered branches to works on ? master ? the branch that will require the less work to integrate ?

Considering the free time I have to work on it, and others projects I work on, I don't expect to finish it before a while. But this is a feature I would really like to have, so I will do the job !

The feature is pretty complex, it will add modes for each tab, like in Simplify3D, it could be view like sub-settings / override, very convenient and would probably drastically reduce number of presets people have to do : 
![image](https://user-images.githubusercontent.com/12198339/169806795-8acf70be-a3f5-48d5-bbc8-5d06e0b1bd69.png)

![image](https://user-images.githubusercontent.com/12198339/169809514-a0b08275-5736-4cdf-a325-e8cc7acba5fc.png)

I started this months ago but I had to do some other jobs since and I just get back on this. The GUI is mainly ok, I have now to work on all the hidden features, wich is pretty complexe at least for me... Presets source code is a jungle :)

**Comment by supermerill** [comment, marked as answer]
How the branches (should) works:

* master is the last release
* stable is the last stable release (often the same as master)
* master_dev is the last beta release, I try to not rewrite commit behind it (seems like I moved it at a wrong place right now, sorry)
* nigthly_dev is the last nightly, it can jump as I may have to rewrite commit below it to make them nicer
* dev: is my dev branch
* merill-merge: it's like master_dev, but on the slic3r flavor (color and names)

I advise you to use 'merill-merge' if there is no problems with it, otherwise master_dev.
If you really need something from the nightly, then use another one.
